### PR TITLE
!feat(client): uds support

### LINKED ISSUE #46: Unix domain socket support
Hello, 
I have a fork of this repo to support sending logs to fluentd server using unix socket files instead of TCP. The change is simple  (commit: https://github.com/johnmanjiro13/tokio-fluent/compare/main...danielsig727:tokio-fluent:uds_support ), and it's tested on my own project.

I'd like to back-port to here. However, I need your opinion on the interface change.
In my fork, for simplicity, I changed `fn new(config: &Config)` to `new_tcp(addr: SocketAddr, config: &Config)` and `new_unix(path: &str, config: &Config)`, and removed `addr` from `Config`. Another way could be keeping `fn new(config: &Config)` and `Config.addr` for backward compatibility, and adding `new_unix(path: &str, config: &Config)` but it neglects the `addr`. What do you think? Thank you

### PR DESCRIPTION
**Pull request by danielsig727**
resolves https://github.com/johnmanjiro13/tokio-fluent/issues/46

### CODE REVIEW

File: README.md
Comment (johnmanjiro13) [inline README.md:1]: Please fix `Setting config values`, too.
https://github.com/johnmanjiro13/tokio-fluent/blob/a059ac9ccf0bfbb6216165d13c2a7a59d3597816/README.md?plain=1#L59-L68

Comment (danielsig727) [reply README.md:1]: fixed. thanks

File: README.md [hunk 1]
```diff
@@ -59,10 +67,10 @@ async fn main() {
 ## Setting config values
 
 ```rust
-let client = Client::new(&Config {
-        addr: "127.0.0.1:24224".parse().unwrap(),
-        ..Default::default()
-    })
+let client = Client::new(
```
Comment (johnmanjiro13) [inline README.md]: 🙏 
```suggestion
let client = Client::new_tcp(
```

Comment (danielsig727) [reply README.md]: oops. thanks

**Review by johnmanjiro13 — APPROVED** [review]
LGTM
Thank you!

**Pull request merged**